x-pack/plugin/logsdb/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeRollingUpgradeIT.java

```diff
@@ -187,40 +158,64 @@ static void enableLogsdbByDefault() throws IOException {
         assertOK(client().performRequest(request));
     }
 
+    private String bulkIndexRequestBody() {
+        final StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < randomIntBetween(10, 20); i++) {
+            sb.append(
+                BULK_INDEX_REQUEST_TEMPLATE
+                    .replace("$timestamp", DateFormatter.forPattern(FormatNames.DATE_TIME.getName()).format(Instant.now()))
+                    .replace("$hostname", randomFrom("potato.host", "tomato.host"))
+                    .replace("$method", randomFrom("PUT", "POST", "GET"))
+                    .replace("$ip", NetworkAddress.format(randomIp(randomBoolean())))
+                    .replace("$message", randomAlphaOfLength(128))
+            );
+            sb.append("\n");
+        }
+        return sb.toString();
+    }
+
     private void assertIndexSettings(int backingIndex, final Matcher<Object> indexModeMatcher) throws IOException {
-        assertThat(
-            getSettings(client(), getWriteBackingIndex(client(), "logs-apache-production", backingIndex)).get("index.mode"),
-            indexModeMatcher
-        );
+        assertThat(getSettings(client(), getWriteBackingIndex(client(), DATA_STREAM, backingIndex)).get("index.mode"), indexModeMatcher);
     }
 
-    private static Request createDataStream(final String dataStreamName) {
-        return new Request("PUT", "/_data_stream/" + dataStreamName);
+    private static void createDataStream() throws IOException {
+        final Request request = new Request("PUT", "/_data_stream/" + DATA_STREAM);
+        final Response response = client().performRequest(request);
+        assertOK(response);
     }
 
-    private static Request bulkIndex(final String dataStreamName, final Supplier<String> bulkIndexRequestSupplier) {
-        final Request request = new Request("POST", dataStreamName + "/_bulk");
+    private static void bulkIndex(final Supplier<String> bulkIndexRequestSupplier) throws IOException {
+        final Request request = new Request("POST", DATA_STREAM + "/_bulk");
         request.setJsonEntity(bulkIndexRequestSupplier.get());
         request.addParameter("refresh", "true");
-        return request;
+
+        final Response response = client().performRequest(request);
+        final var responseBody = entityAsMap(response);
+
+        // then - ensure no issues
+        assertOK(response);
+        assertThat("errors in response:\n " + responseBody, responseBody.get("errors"), Matchers.is(false));
```

Comment:
printing the error in response here makes finding the errors significantly easier. Previously, I had to log dive to understand what happened